src/server/tls.rs

```diff
@@ -56,26 +56,26 @@ impl TryFrom<FtpsConfig> for tokio_rustls::TlsAcceptor {
     }
 }
 
-pub fn new_config<P: AsRef<Path>>(certs_file: P, key_file: P) -> Arc<rustls::ServerConfig> {
-    let certs: Vec<Certificate> = load_certs(certs_file);
-    let privkey: PrivateKey = load_private_key(key_file);
+pub fn new_config<P: AsRef<Path>>(certs_file: P, key_file: P) -> std::io::Result<Arc<rustls::ServerConfig>> {
+    let certs: Vec<Certificate> = load_certs(certs_file)?;
+    let privkey: PrivateKey = load_private_key(key_file)?;
 
     let mut config = rustls::ServerConfig::new(NoClientAuth::new());
     config.session_storage = Arc::new(rustls::NoServerSessionStorage {});
     config.key_log = Arc::new(rustls::KeyLogFile::new());
     config.set_single_cert(certs, privkey).expect("Failed to setup TLS certificate chain and key");
-    Arc::new(config)
+    Ok(Arc::new(config))
 }
 
-fn load_certs<P: AsRef<Path>>(filename: P) -> Vec<rustls::Certificate> {
-    let certfile: File = File::open(filename).expect("cannot open certificate file");
+fn load_certs<P: AsRef<Path>>(filename: P) -> std::io::Result<Vec<rustls::Certificate>> {
+    let certfile: File = File::open(filename)?;
     let mut reader: BufReader<File> = BufReader::new(certfile);
-    rustls::internal::pemfile::certs(&mut reader).unwrap()
+    Ok(rustls::internal::pemfile::certs(&mut reader).unwrap())
 }
 
-fn load_private_key<P: AsRef<Path>>(filename: P) -> rustls::PrivateKey {
+fn load_private_key<P: AsRef<Path>>(filename: P) -> std::io::Result<rustls::PrivateKey> {
```

Comment:
Suggestion(non-blocking): see above `use rustls::PrivateKey`

Reply:
I agree and fixed it. Kept `std::io::Result`